### DISCUSSION TITLE
RPC responded Insufficient funds even if i have funds

### DISCUSSION THREAD

**Discussion by DerXanRam** [mhw0/libethc#30] [Q&A]
I'm trying to swap on Binance Smart Chain(which this library also works for it) using 1inch dex. But it says sometimes 
```"message":"insufficient funds for gas * price + value"```
and some times
```"message":"INTERNAL_ERROR: insufficient funds"```

even if i have fund in my account to do the swap and cover the gas.  all my balance is BNB and i'm swaping BNB(0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee) to BUSD(0xe9e7cea3dedca5984780bafc599bd69add087d56).

this are the values used in making the raw tnx
```
nonce=0x1
gasPrice is 0xd09dc300
GasLimit is 0x42c23
value is 0x2386f26fc10000
ToAddress is 0x1111111254eeb25477b68fb85ed929f73a960582
```
The ABI data is 

`0x12aa3caf000000000000000000000000f8a1527325bb62d6d0320eb2312748d590f1ce27000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee000000000000000000000000e9e7cea3dedca5984780bafc599bd69add087d56000000000000000000000000f8a1527325bb62d6d0320eb2312748d590f1ce27000000000000000000000000b438d71c1e6e8e53637e1f1dec37c2e635e335fc000000000000000000000000000000000000000000000000002386f26fc100000000000000000000000000000000000000000000000000001cfbbfaad4cc354a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000014000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000016800000000000000000000000000000000000000000000000000014a00001a4041bb4cdb9cbd36b01bd1cbaebf2de08d9173bc095cd0e30db05120830c9e0295730bbb220bd78346c6499ff37851fbbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c00a4a5dcbcdf000000000000000000000000bb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c000000000000000000000000e9e7cea3dedca5984780bafc599bd69add087d560000000000000000000000002c93437fa01c0123ed21e8b862762c887c1195130000000000000000000000001111111254eeb25477b68fb85ed929f73a960582ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008b1ccac8`

and this is the raw tnx generated
`f9036f0184d09dc30083042c23941111111254eeb25477b68fb85ed929f73a96058280b9030812aa3caf000000000000000000000000f8a1527325bb62d6d0320eb2312748d590f1ce27000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee000000000000000000000000e9e7cea3dedca5984780bafc599bd69add087d56000000000000000000000000f8a1527325bb62d6d0320eb2312748d590f1ce27000000000000000000000000b438d71c1e6e8e53637e1f1dec37c2e635e335fc000000000000000000000000000000000000000000000000002386f26fc100000000000000000000000000000000000000000000000000001cfbbfaad4cc354a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000014000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000016800000000000000000000000000000000000000000000000000014a00001a4041bb4cdb9cbd36b01bd1cbaebf2de08d9173bc095cd0e30db05120830c9e0295730bbb220bd78346c6499ff37851fbbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c00a4a5dcbcdf000000000000000000000000bb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c000000000000000000000000e9e7cea3dedca5984780bafc599bd69add087d560000000000000000000000002c93437fa01c0123ed21e8b862762c887c1195130000000000000000000000001111111254eeb25477b68fb85ed929f73a960582ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008b1ccac88193a0e67028be55afd7e2d967d969b1ca6ad19cc87b9a4157701c12704675fa21d560a06767242acafe600388c1909d0b599417f85937c0c069b9c6dddbf5075d070010`

any encoding error ?Please help :pray:

**Comment by DerXanRam** [comment]
this is the code snippet i use to encode the transaction

```
ok(eth_rlp_init(&rlp0, ETH_RLP_ENCODE));
		ok(eth_rlp_array(&rlp0));
		ok(eth_rlp_uint8(&rlp0, &nonce));
		ok(eth_rlp_hex(&rlp0, &gasprice, NULL));
		ok(eth_rlp_hex(&rlp0, &gaslimit, NULL));
		ok(eth_rlp_address(&rlp0, &toaddr));
		ok(eth_rlp_uint8(&rlp0, &zero));
		ok(eth_rlp_hex(&rlp0, &abi_result, NULL));
		ok(eth_rlp_uint16(&rlp0, &chainid));
		ok(eth_rlp_uint8(&rlp0, &zero)); //   0x,
		ok(eth_rlp_uint8(&rlp0, &zero)); //   0x,
		ok(eth_rlp_array_end(&rlp0));	 // ]

		ok(eth_rlp_to_bytes(&rlp0bytes, &rlp0len, &rlp0));
		ok(eth_rlp_free(&rlp0));

		// compute the keccak hash of the encoded rlp elements
		ok(eth_keccak256(keccak, rlp0bytes, rlp0len));
		free(rlp0bytes);

		// sign the transaction
		ok(eth_ecdsa_sign(&sign, privkey, keccak));

		// calculate v
		v = sign.recid + chainid * 2 + 35;
		r = sign.r;
		s = sign.s;

		ok(eth_rlp_init(&rlp1, ETH_RLP_ENCODE));
		ok(eth_rlp_array(&rlp1));
		ok(eth_rlp_uint8(&rlp1, &nonce));
		ok(eth_rlp_hex(&rlp1, &gasprice, NULL));
		ok(eth_rlp_hex(&rlp1, &gaslimit, NULL));
		ok(eth_rlp_address(&rlp1, &toaddr));
		ok(eth_rlp_uint8(&rlp1, &zero));
		ok(eth_rlp_hex(&rlp1, &abi_result, NULL)); // replaced
		ok(eth_rlp_uint8(&rlp1, &v));
		ok(eth_rlp_bytes(&rlp1, &r, &siglen));
		ok(eth_rlp_bytes(&rlp1, &s, &siglen));
		ok(eth_rlp_array_end(&rlp1));

		// FIX3: this actually returns the output length
		ok(eth_rlp_to_hex(&txn, &rlp1) > 0);
		ok(eth_rlp_free(&rlp1));
```

**Comment by mhw0** [comment]
Opened an issue: #31
